### ISSUE TITLE
Seem like char_feat broke the program, perhaps sharing char_feat and word_feat while training using disjoint multitasks mode

### ISSUE THREAD

**Issue by puttkraidej** [facebookresearch/pytext#267]
## Steps to reproduce

  1. my pytext repo cloned from `commit 5d0e6af0092b7cb02cbc873e9733e47729b1e982
Author: Haoran Li <aimeeli@fb.com>
Date:   Thu Jan 10 18:13:00 2019 -0800`
  2. `$pytext train < myconfigfile`
  3. I got this error message
`Traceback (most recent call last):
  File "/home/jod/pytext/pytext_venv/bin/pytext", line 11, in <module>
    load_entry_point('pytext-nlp', 'console_scripts', 'pytext')()
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/main.py", line 266, in train
    train_model(config, summary_writer=summary_writer)
  File "/home/jod/pytext/pytext/pytext/workflow.py", line 69, in train_model
    config, dist_init_url, device_id, rank, world_size, summary_writer, metadata
  File "/home/jod/pytext/pytext/pytext/workflow.py", line 97, in prepare_task
    task = create_task(config.task, metadata=metadata)
  File "/home/jod/pytext/pytext/pytext/task/task.py", line 38, in create_task
    return create_component(ComponentType.TASK, task_config, metadata, model_state)
  File "/home/jod/pytext/pytext/pytext/config/component.py", line 142, in create_component
    return cls.from_config(config, *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/task/disjoint_multitask.py", line 57, in from_config
    data_handler.init_metadata()
  File "/home/jod/pytext/pytext/pytext/data/disjoint_multitask_data_handler.py", line 150, in init_metadata
    data_handler.init_metadata()
  File "/home/jod/pytext/pytext/pytext/data/data_handler.py", line 351, in init_metadata
    self.init_metadata_from_path(self.train_path, self.eval_path, self.test_path)
  File "/home/jod/pytext/pytext/pytext/data/data_handler.py", line 343, in init_metadata_from_path
    for path in [train_path, eval_path, test_path]
  File "/home/jod/pytext/pytext/pytext/data/data_handler.py", line 365, in _init_metadata
    self.init_feature_metadata(train_data, eval_data, test_data)
  File "/home/jod/pytext/pytext/pytext/data/data_handler.py", line 398, in init_feature_metadata
    feat.build_vocab(*vocab_data, min_freq=feat.min_freq)
  File "/home/jod/pytext/pytext/pytext/fields/char_field.py", line 52, in build_vocab
    for word_chars in sentence:
TypeError: 'NoneType' object is not iterable`

## Observed Results

  * By the way, in my opinion
`  File "/home/jod/pytext/pytext/pytext/fields/char_field.py", line 52, in build_vocab`
`     for word_chars in sentence:`
`TypeError: 'NoneType' object is not iterable`
is the main error here.

Here is my config for all multitasks (composed of 1 LM task and 3 DocClassification tasks). I have 
configured as
`"char_feat": {
                    "embed_dim": 100,
                    "sparse": false,
                    "cnn": {
                        "kernel_num": 100,
                        "kernel_sizes": [
                            3,
                            4
                        ]
                    },
                    "export_input_names": [
                        "char_vals"
                    ],
                    "vocab_from_train_data": true
                },
                "word_feat": {
                    "embed_dim": 100,
                    "vocab_file": "myvocabfile",
                    "vocab_size": 30000,
                  "vocab_from_train_data": false
                }`

for every tasks.

## Expected Results

  I expect every tasks will share char and word embedding modules, because I have done this type of sharing before using pytext but only word-wise level. I might did something wrong, any suggestion would be appreciated.

my input data for (binary)DocClassification similars to
__label__0\tBOS word1 word2 word99 EOS
__label__1\tBOS word5 word77 word8 word4 EOS

and

BOS word82 word32 word76 EOS
BOS word22 word67 word80 word59 EOS

for LM task.

Thanks in advance

**Comment by puttkraidej** [comment]
FYI
I also tried the most recent commit
`commit 70c1b4361eed09b11c32a8c4a1def8807e573351
Author: Barlas Oguz <barlaso@fb.com>
Date:   Mon Feb 4 11:01:33 2019 -0800

    export target model to default path

    Summary: Pull Request resolved: https://github.com/facebookresearch/pytext/pull/264

    Differential Revision: D13933847

    fbshipit-source-id: 40a0c576d729d6c2ce1b6c7028ecaabcc42f19d7`

I found this error instead
`Traceback (most recent call last):
  File "/home/jod/pytext/pytext_venv/bin/pytext", line 11, in <module>
    load_entry_point('pytext-nlp', 'console_scripts', 'pytext')()
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/jod/pytext/pytext_venv/lib/python3.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/main.py", line 272, in train
    train_model(config, summary_writer=summary_writer)
  File "/home/jod/pytext/pytext/pytext/workflow.py", line 69, in train_model
    config, dist_init_url, device_id, rank, world_size, summary_writer, metadata
  File "/home/jod/pytext/pytext/pytext/workflow.py", line 97, in prepare_task
    task = create_task(config.task, metadata=metadata)
  File "/home/jod/pytext/pytext/pytext/task/task.py", line 38, in create_task
    return create_component(ComponentType.TASK, task_config, metadata, model_state)
  File "/home/jod/pytext/pytext/pytext/config/component.py", line 144, in create_component
    return cls.from_config(config, *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/task/disjoint_multitask.py", line 93, in from_config
    for name, task in task_config.tasks.items()
  File "/home/jod/pytext/pytext/pytext/task/disjoint_multitask.py", line 93, in <genexpr>
    for name, task in task_config.tasks.items()
File "/home/jod/pytext/pytext/pytext/config/component.py", line 164, in create_model
    return create_component(ComponentType.MODEL, model_config, *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/config/component.py", line 144, in create_component
    return cls.from_config(config, *args, **kwargs)
  File "/home/jod/pytext/pytext/pytext/models/language_models/lmlstm.py", line 79, in from_config
    model = super().from_config(model_config, feat_config, metadata)
  File "/home/jod/pytext/pytext/pytext/models/model.py", line 161, in from_config
    feat_config, create_fn=cls.create_embedding, metadata=metadata
  File "/home/jod/pytext/pytext/pytext/models/module.py", line 44, in create_module
    create_fn(module_config, *args, **kwargs),
  File "/home/jod/pytext/pytext/pytext/models/model.py", line 151, in create_embedding
    sub_emb_module_dict = cls.create_sub_embs(feat_config, metadata)
  File "/home/jod/pytext/pytext/pytext/models/model.py", line 123, in create_sub_embs
    config, metadata=meta
KeyError: 'char_feat'`

again I think the main error should be `KeyError: 'char_feat'`.

Any idea?

**Comment by puttkraidej** [comment]
It's my fault, I force the LM task to have char_feat which I found out later that it's not available (correct me if I'm wrong)
I employed char_feat to single Doc classification task and it works smoothly. By the way, when it comes to multitask (multiple doc classification tasks) I got this error
`RuntimeError: CUDA error: device-side assert triggered`
quickly guess I just have had my own resource allocation issue.
Thanks and sorry if this issue opening disturb you.
